Makefile

```diff
@@ -3,12 +3,36 @@
 VERSION ?= 0.0.1.Final-SNAPSHOT
 COMMIT_HASH ?= $(shell git rev-parse HEAD)
 
+# Version label is used in the OpenShift/K8S resources to identify
+# their specific instances. These resources will have labels of
+# "app: kiali" and "version: ${VERSION_LABEL}" In this development
+# environment, setting the label names equal to the branch names
+# allows developers to build and deploy multiple Kiali instances
+# at the same time which is useful for debugging and testing.
+# Due to restrictions on allowed characters, we convert
+# uppercase characters to lowercase characters.
+# If we are deploying from a branch, we must ensure all the OS/k8s names to be created
+# are unique - the NAME_SUFFIX will be appended to all names so they are unique.
+VERSION_LABEL ?= $(shell git rev-parse --abbrev-ref HEAD | tr '[:upper:]' '[:lower:]')
```

Comment:
Aren't there other restrictions on "bad characters" like underscore?

Reply:
The rules are as follows, and yes it will fail if your branch contains an invalid character (such as an '_'):

a DNS-1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')